apps-rendering/src/pagination.ts

```diff
@@ -0,0 +1,146 @@
+import { map, none, some, withDefault } from '@guardian/types';
+import type { Option } from '@guardian/types';
+import { pipe } from 'lib';
+import type { LiveBlock } from 'liveBlock';
+
+export type PageReference = {
+	blocks: LiveBlock[];
+	pageNumber: number;
+	suffix: string;
+};
+
+export type Pagination = {
+	newest: Option<string>;
+	newer: Option<string>;
+	oldest: Option<string>;
+	older: Option<string>;
+	numberOfPages: number;
+};
+
+export type LiveBlogPagedBlocks = {
+	currentPage: PageReference;
+	pagination: Pagination;
+};
+
+const navContainerId = 'liveblog-navigation';
+
+const getOldestPage = (
+	pages: LiveBlock[][],
+	pageNumber: number,
+): Option<string> => {
+	if (pageNumber >= pages.length) return none;
+
+	const blocks = pages[pages.length - 1];
+	return some(`?page=with:block-${blocks[0].id}#${navContainerId}`);
+};
+
+const getOlderPage = (
+	pages: LiveBlock[][],
+	pageNumber: number,
+): Option<string> => {
+	if (pageNumber < pages.length) {
+		const blocks = pages[pageNumber];
+		return some(`?page=with:block-${blocks[0].id}#${navContainerId}`);
+	}
+
+	return none;
+};
+
+const getNewerPage = (
+	pages: LiveBlock[][],
+	pageNumber: number,
+): Option<string> => {
+	if (pageNumber > 2) {
+		const blocks = pages[pageNumber - 2];
+		return some(`?page=with:block-${blocks[0].id}#${navContainerId}`);
+	}
+
+	if (pageNumber === 2) {
+		return some(`?`);
+	}
+
+	return none;
+};
+
+const getNewestPage = (
+	pages: LiveBlock[][],
+	pageNumber: number,
+): Option<string> => {
+	if (pageNumber > 1) {
+		return some(`?`);
+	}
+
+	return none;
+};
+
+const getFirstPageLength = (pageSize: number, blocks: LiveBlock[]): number => {
```

Comment:
We want the final page of a liveblog to always equal the designated page size (either 10 or 30 blocks). For example, if we have a page size of 10 and 21 blocks in the liveblog, we don't want 3 pages (10 blocks, 10 blocks, 1 block) we only want 2 pages (11 blocks, 10 blocks).


As such, we need the length of blocks in the first page to flex.